miso-web/src/main/webapp/scripts/workflow_display.js

```diff
@@ -0,0 +1,167 @@
+      id: workflowId
+    })), showSuccess, showError);
+  }
+
+  function showError(xhr) {
+    loadingTag.remove();
+    display.children().show();
+    display.children("input").focus();
+    errorDiv.text(parseError(xhr));
+  }
+
+  function parseError(xhr) {
+    var responseObj = JSON.parse(xhr.responseText);
+    if (responseObj.detail && responseObj.dataFormat === "validation") {
+      return responseObj["data"]["GENERAL"];
+    }
+
+    return responseObj["message"];
+  }
+
+  function processInput(workflowId, stepNumber, input) {
+    ajax("POST", encodeURI("/miso/rest/workflow/process/?" + jQuery.param({
+      id: workflowId,
+      stepNumber: stepNumber,
+      input: input
+    })), updateDisplay, showError);
+  }
+
+  function makeMessageTag(message) {
+    return jQuery("<p>" + message + "</p>");
+  }
+
+  function showLoading() {
+    // Hide children in case the request fails and we need to show them later
+    display.children().hide();
+    display.append(loadingTag);
+  }
+
+  function makeInputTag(workflowId, stepNumber) {
+    var inputTag = jQuery("<input type='text'>");
+
+    var doProcess = function() {
+      processInput(workflowId, stepNumber, inputTag.val())
+    };
+    inputTag.keypress(function(e) {
+      if (e.which === 13) {
+        doProcess();
+      }
+    }).bind("paste", function() {
+      setTimeout(doProcess, 100);
+    });
+
+    return inputTag;
+  }
+
+  function changeStep(workflowId, stepNumber) {
+    ajax("GET", encodeURI("/miso/rest/workflow/changestep/?" + jQuery.param({
+      id: workflowId,
+      stepNumber: stepNumber
+    })), updateDisplay, showError);
+  }
+
+  function cancelInput(workflowId) {
+    ajax("POST", "/miso/rest/workflow/cancelinput/?" + jQuery.param({
+      id: workflowId
+    }), updateDisplay, showError);
+  }
+
+  function makeArrowCell(entryStepNumber, currentStepNumber) {
+    return entryStepNumber === currentStepNumber ? jQuery("<td>").append(jQuery("<img src='/styles/images/arrow.svg' class='logIcon'>"))
+        : jQuery("<td>");
+  }
+
+  function makeModifyCell(entryStepNumber, numEntries) {
+    return entryStepNumber === numEntries - 1 ? jQuery("<td>").append(jQuery("<img src='/styles/images/cancel.png' class='logIcon'>"))
+        : jQuery("<td>").append(jQuery("<img src='/styles/images/redo.svg' class='logIcon'>"));
+  }
+
+  function makeLogEntryClickHandler(entryStepNumber, numEntries, workflowId) {
+    return function() {
+      entryStepNumber === numEntries - 1 ? cancelInput(workflowId) : changeStep(workflowId, entryStepNumber)
+    };
+  }
+
+  function makeLog(logEntries, workflowId, currentStepNumber) {
+    var table = jQuery("<table class='workflowLogTable'>");
+
+    for (var i = 0; i < logEntries.length; i++) {
+      table.prepend(jQuery("<tr>").append(
+          [makeArrowCell(i, currentStepNumber), jQuery("<td>").text(logEntries[i]), jQuery("<td>"), makeModifyCell(i, logEntries.length)])
```

Comment:
Is there a blank column in here between the log message and action button?

Reply:
There is, good catch.  I put it there to balance the "arrow" column that precedes the log message.  Both the "arrow" column and the "blank" column will have the same width, as dictated by CSS, but I'm open to suggestions.